include/seastar/core/temporary_buffer.hh

```diff
@@ -34,6 +34,43 @@
 
 namespace seastar {
 
+/*
+ * The custom-deleter constructor of temporary_buffer has been made private to
+ * better ensure the safety of cross-shard sharing the temporary_buffer.
+ *
+ * The forward declarations here are to allow for some manually verified, pre-existing,
+ * code to still access the newly-private constructor.
+ */
+
+namespace net {
+class packet;
+class packet_data_source;
+
+template <typename Protocol>
+class native_connected_socket_impl;
```

Comment:
I guess if any of these go stale compilation will just fail so there is no danger there?

It's all a bit ugly but I guess it is what it is and we can just see how it will work out